src/bidiMapper/domains/script/Realm.ts

```diff
@@ -833,16 +735,256 @@ export class Realm {
+          `SharedId "${localValue.sharedId}" was not found.`
+        );
+      }
+      const {documentId, backendNodeId} = parsedSharedId;
+      // TODO: add proper validation if the element is accessible from the current realm.
+      if (this.browsingContext.navigableId !== documentId) {
+        throw new NoSuchNodeException(
+          `SharedId "${localValue.sharedId}" belongs to different document. Current document is ${this.browsingContext.navigableId}.`
+        );
+      }
+
+      try {
+        const {object} = await this.cdpClient.sendCommand('DOM.resolveNode', {
+          backendNodeId,
+          executionContextId: this.executionContextId,
+        });
+        // TODO(#375): Release `obj.object.objectId` after using.
+        return {objectId: object.objectId};
+      } catch (error: any) {
+        // Heuristic to detect "no such node" exception. Based on the  specific
+        // CDP implementation.
+        if (
+          error.code === CdpErrorConstants.GENERIC_ERROR &&
+          error.message === 'No node with given id found'
+        ) {
+          throw new NoSuchNodeException(
+            `SharedId "${localValue.sharedId}" was not found.`
+          );
+        }
+        throw new UnknownErrorException(error.message, error.stack);
+      }
+    }
+    return await super.deserializeForCdp(localValue);
+  }
+
+  override async evaluate(
+    expression: string,
+    awaitPromise: boolean,
+    resultOwnership: Script.ResultOwnership,
+    serializationOptions: Script.SerializationOptions,
+    userActivation?: boolean
+  ): Promise<Script.EvaluateResult> {
+    await this.#browsingContextStorage
+      .getContext(this.#browsingContextId)
+      .targetUnblockedOrThrow();
+
+    return await super.evaluate(
+      expression,
+      awaitPromise,
+      resultOwnership,
+      serializationOptions,
+      userActivation
+    );
+  }
+
+  override async callFunction(
+    functionDeclaration: string,
+    thisLocalValue: Script.LocalValue,
+    argumentsLocalValues: Script.LocalValue[],
+    awaitPromise: boolean,
+    resultOwnership: Script.ResultOwnership,
+    serializationOptions: Script.SerializationOptions,
+    userActivation?: boolean
+  ): Promise<Script.EvaluateResult> {
+    await this.#browsingContextStorage
+      .getContext(this.#browsingContextId)
+      .targetUnblockedOrThrow();
+
+    return await super.callFunction(
+      functionDeclaration,
+      thisLocalValue,
+      argumentsLocalValues,
+      awaitPromise,
+      resultOwnership,
+      serializationOptions,
+      userActivation
+    );
+  }
+}
+
+export class DedicatedWorkerRealm extends Realm {
+  readonly #ownerRealm: WindowRealm | DedicatedWorkerRealm;
+
+  constructor(
+    cdpClient: CdpClient,
+    eventManager: EventManager,
+    executionContextId: Protocol.Runtime.ExecutionContextId,
+    logger: LoggerFn | undefined,
+    origin: string,
+    ownerRealm: WindowRealm | DedicatedWorkerRealm,
```

Comment:
Why cannot it be just any realm?

Reply:
It is generally only `WindowRealm` and `DedicatedWorkerRealm`, but I guess the distinction doesn't matter.